internal/otel/config/helpers/processors/batch_processor.go

```diff
@@ -16,11 +16,14 @@ const batchProcessorName = "batch"
 var BatchProcessorID component.ID = component.NewID(batchProcessorName)
 
 // BatchProcessorCfg  generates the config for a batch processor.
-func BatchProcessorCfg() *batchprocessor.Config {
+func BatchProcessorCfg(timeout time.Duration) *batchprocessor.Config {
+	if timeout == 0 {
```

Comment:
Should we do this earlier when we set the params?

Reply:
After discussing, we agreed we should do the validation in the `newProvider` layer, or before that!